lib/if.h

```diff
@@ -19,6 +19,20 @@ extern "C" {
 
 DECLARE_MTYPE(CONNECTED_LABEL);
 
+/* Interface type - ones of interest. */
+enum intf_type {
+	IF_OTHER = 0,  /* Anything else */
+	IF_VXLAN,      /* VxLAN interface */
+	IF_VRF,	       /* VRF device */
+	IF_BRIDGE,     /* bridge device */
+	IF_VLAN,       /* VLAN sub-interface */
+	IF_MACVLAN,    /* MAC VLAN interface*/
+	IF_VETH,       /* VETH interface*/
+	IF_BOND,       /* Bond */
+	IF_BOND_SLAVE, /* Bond Slave*/
```

Comment:
There's no such interface type as BOND_SLAVE.

Reply:
Overall, as you directly assign this type from `enum zebra_iftype`, I think it's much better to move `enum zebra_iftype` to a library and use a single type in lib and zebra. Otherwise there's a huge chance that the types will become out of sync.

Reply:
Sure, will remove the BOND_SLAVE. 

As to moving/consolidating the enum zebra_iftype at one place in if.h, it involves lots of files to be touched. The work is in progress and i will raise a subsequent MR to do that work.

Reply:
It's used in just 5 files, all of them in zebra. I don't see any reason to merge this as-is, instead of using a shared enum from lib/if.h everywhere.

Reply:
Sure, let me work on a cleaner solution and upload the changes